release-notes/24.05/index.html

```diff
@@ -0,0 +1,80 @@
+  </head>
+  <body>
+    <main role="main">
+      <section class="jumbotron text-left rounded-0 m-0">
+  <div class="container">
+<h1>CheriBSD 23.11 Release Notes</h1>
+<h2>Overview</h2>
+<p>
+Welcome to the Spring 2024 (24.05) release of the CheriBSD operating system.  As well as a general update of the baseline FreeBSD OS (corresponding to FreeBSD CURRENT as of January 2024) from which CheriBSD is derived, we have introduced several new research components which will be of interest to the CHERI and CheriBSD community:
+</p>
+<ul>
+<li>Library Compartmentalization (<a href="https://man.cheribsd.org/cgi-bin/man.cgi/release-24.05/c18n">https://man.cheribsd.org/cgi-bin/man.cgi/release-24.05/c18n</a>), is increasingly mature and enabled by default for CheriABI and benchmark ABI processes. While this implementation is sufficently functional to enable by default, it is neither complete nor intended to be secure.
+<li>To strengthen library compartmentalization, we have completed an adjustment to the ABI for functions whose arguments do not fit in registers.  These arguments are now reachable via a bounded capability passed in register C9 and newly compiled callees access them via C9 allowing callers to place argument values anywhere they desire.  This is incompatible with callers compiled for releases prior to 23.11 (which used a transitional ABI setting C9 in the caller, but not using it in the callee).
+<li>User level heap temporal safety performance has been improved by adding in-kenrel worker threads.  This change allows userspace threads to invoke the kernel revoker, proceed with work, and then come back later to complete revocation.
+<li>Our CHERI-aware port of the bhyve hypervisor has been synced with AArch64 support merged to upstream FreeBSD.  The GDB stub has been enhanced to support reading capabilities in guest memory.
+<li>Kernels may now be compiled with the pure-capability benchmark ABI in addition to the existing hybrid and pure-capability (CheriABI) ABIs. This ABI uses a modified form of code generation which improves performance on the Arm Morello board, required due to limitations with bounds prediction in the current Morello prototype. The performance of the benchmark ABI is more predictive of potential future CHERI microarchitectural performance than the Morello prototype running software compiled for CheriABI, making it preferable for benchmarking. However, this comes at the cost of reduced security, and so software compiled for the benchmark ABI should not be used for security evaluation. For details on how to use the benchmark ABI, and benchmark Morello in general, readers should see the Getting Started Guide.
+<li>The <tt>MAP_CHERI_NOSETBOUNDS</tt> flag to the mmap system call is now deprecated and a no-op.  When mmap is falled with the <tt>MAP_FIXED</tt> flag and an address pointing to an existing reservation, the passed address will be returned unmodified on success as if <tt>MAP_CHERI_NOSETBOUNDS</ttt> had been passed.
```

Comment:
```suggestion
<li>The <tt>MAP_CHERI_NOSETBOUNDS</tt> flag to the mmap system call is now deprecated and a no-op.  When mmap is called with the <tt>MAP_FIXED</tt> flag and a pointer to an existing reservation, the passed pointer will be returned unmodified on success as if <tt>MAP_CHERI_NOSETBOUNDS</ttt> had been passed.
```

Reply:
I also don't like the "as if" at the end, because the flag doesn't really exist any more. What you really mean is it's the same behaviour as you would have got with MAP_CHERI_NOSETBOUNDS in prior releases.